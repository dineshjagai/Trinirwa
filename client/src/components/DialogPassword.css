.btn {
    color: #00695c;
    border: none; 
    background-color: inherit;
}

#delete:hover {
    background: white;
}