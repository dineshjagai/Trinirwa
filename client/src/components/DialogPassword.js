import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { Redirect } from 'react-router-dom';
import { deactivateProfile } from './Module';
import './DialogPassword.css';
import { getCurrentUsername } from '../auth/authServices';

export default function DialogPassword() {
  const username = getCurrentUsername();
  const [open, setOpen] = useState(false);
  const [redirect, setRedirect] = useState(false);
  const [clue, setClue] = useState('');
  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
    setClue('');
  };

  const handleCloseDelete = () => {
    const input = document.getElementById('passwordInput').value;
    if (input !== '') {
      deactivateProfile(username, input).then((response) => {
        console.log(response);
        setOpen(false);
        setRedirect(true);
      }).catch((err) => {
        console.log(err);
        setClue('Incorrect entry');
      });
    }
  };

  return !redirect ? (
    <div>
      <button id="delete" style={{ color: '#00695c', border: 'none', backgroundColor: 'inherit' }} type="button" onClick={handleClickOpen}>Delete account</button>
      <Dialog open={open} onClose={handleClose} aria-labelledby="form-dialog-title">
        <DialogTitle id="form-dialog-title">Confirm account deletion</DialogTitle>
        <DialogContent>
          <TextField
            error={clue !== ''}
            autoFocus
            margin="dense"
            id="passwordInput"
            label="Input password"
            type="password"
            fullWidth
            helperText={clue}
          />
        </DialogContent>
        <DialogActions>
          <button className="btn cancel" type="button" onClick={handleClose}>
            Cancel
          </button>
          <button className="btn delete" type="button" onClick={handleCloseDelete}>
            Delete
          </button>
        </DialogActions>
      </Dialog>
    </div>
  ) : <Redirect to="/login" />;
}
