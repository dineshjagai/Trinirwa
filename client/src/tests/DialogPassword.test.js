import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import DialogPassword from '../components/DialogPassword';
import { render } from '@testing-library/react';

test('Test DialogPassword snapshot', () => {
  const component = renderer.create(<DialogPassword />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

