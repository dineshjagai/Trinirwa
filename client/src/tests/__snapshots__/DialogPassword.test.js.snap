// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DialogPassword snapshot 1`] = `
<div>
  <button
    id="delete"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "border": "none",
        "color": "#00695c",
      }
    }
    type="button"
  >
    Delete account
  </button>
</div>
`;
